geps/gep-1619.md

```diff
@@ -159,47 +264,398 @@ Generally, the implementation API programs the dataplane API; however these two
 
 Java application servers such as Tomcat and Jetty, were the first to standardize the API around persistent sessions. These Java applications introduced the “jsessionid” cookie as well as more advanced features such as session migration, replication, and on demand session activation. It’s important for Gateway API to examine persistent session use cases and history from Java APIs to ensure the API is designed appropriately.
```

Comment:
As far as I know the Java jsessionid cookie was used only to exchange the session information between the client and the server (backend) but not for routing the request to the correct backend (which is the use-case for Gateway API)

Reply:
Agreed. Gateway API (or proxies) introduced the concept of a gateway or proxy which uses cookies to maintain persistent connections. That idea wasn't really applicable to the use cases for jsessionid cookie.

I suppose this language should be tweaked to make sure I'm not conflating the two. "Persistent Session" feel ambiguous sometimes, but in this GEP, I think I usually use it to mean "consistent loadbalancing".

Updated to "cookies" and "sessions" with Java vs. "persistent sessions".

Reply:
Not sure where the Gateway API is introducing this concept. Java HttpSession when used with load balancers is based on a requirement for LB to implement 'strong affinity'. Same does PHP and probably many others. The requirements are the same: "once a backend handles a request from a client, all subsequent requests from the client will go to the same backend". 

I don't think anything changed in this requirement - in the end we still have a browser making a connection that can get sent to a K8S backend ( which may be PHP or Java ) and user will expect it to work as it did before K8S and defined by the language/libraries. 

Incidentally K8S native affinity ( in Service ) seems to also satisfy this requirement. 

I agree that 'persistent sessions' are bigger feature than session affinity - it may involve session migration, has ties to SSO/login, etc - and generally it's part of the language/framework on the backend. 

The goal of this GEP is to support backends implementing persistent sessions ( or other cases of session affinity ), not to introduce new concepts.